ddtrace/profiling/profiler.py

```diff
@@ -132,6 +133,7 @@ class _ProfilerInstance(service.Service):
     _recorder = attr.ib(init=False, default=None)
     _collectors = attr.ib(init=False, default=None)
     _scheduler = attr.ib(init=False, default=None)
+    _lambda_function_name = attr.ib(factory=lambda: os.environ.get("AWS_LAMBDA_FUNCTION_NAME"), type=Optional[str])
```

Comment:
I'd set `init=False` unless you want the user to be able to override it (I don't know if overriding is a valid use case).
If you do want to be able to override it, remove the private `_` prefix.